### PR TITLE
TD-3671 Fixed enrolling from 'Tracking system-Manage Delegate' view doesn't appear on 'Current activities'

### PR DESCRIPTION
**Pull request by sherif-olaboye**
### JIRA link
https://hee-tis.atlassian.net/browse/TD-3671

### Description
Alter the store procedure and function that select available and current activities courses
Also implement the new business rule to change the value of EnrollmentMethodID  to current when the course is enrol
### Screenshots
image

-----
### Developer checks
(Leave tasks unticked if they haven't been appropriate for your ticket.)

I have:
- [x] Run the formatter and made sure there are no IDE errors (see [info on Text Editor settings](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546185813/DLS+Dev+Process) to avoid whitespace changes)
- [ ] Written tests for the changes (accessibility tests, unit tests for controller, data services, services, view models, etc)
- [x] Manually tested my work with and without JavaScript
- [ ] Tested any Views or partials created or changed with [Wave Chrome plugin](https://chrome.google.com/webstore/detail/wave-evaluation-tool/jbbplnpkjmmeebjpijfedlgcdilocofh/related) and addressed any valid accessibility issues
- [ ] Updated/added documentation in [Confluence](https://hee-tis.atlassian.net/wiki/spaces/TP/pages/3546939432/DLS+Code) and/or [GitHub Readme](https://github.com/TechnologyEnhancedLearning/DLSV2/blob/master/README.md). List of documentation links added/changed:
  - [doc_1_here](link_1_here)
- [ ] Updated my Jira ticket with information about other parts of the system that were touched as part of the MR and have to be sanity tested to ensure nothing’s broken
- [x] Scanned over my pull request in GitHub and addressed any warnings from the GitHub Build and Test checks.

### CODE REVIEW

**Comment by rshrirohit** [comment]
The function and SP changes are correct. For the update, we don't need to change EnrollmentMethodID so let's remove it.
Also, check the scenario where we show a list of available courses while enrolling a delegate to a course.

File: DigitalLearningSolutions.Data.Migrations/DigitalLearningSolutions.Data.Migrations.csproj [hunk 1]
```diff
@@ -35,7 +35,7 @@
 
   <ItemGroup>
     <EmbeddedResource Update="Properties\Resources.resx">
-      <Generator>ResXFileCodeGenerator</Generator>
+      <Generator>PublicResXFileCodeGenerator</Generator>
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data.Migrations/DigitalLearningSolutions.Data.Migrations.csproj:38]: Why is this change needed?

Comment (sherif-olaboye) [reply DigitalLearningSolutions.Data.Migrations/DigitalLearningSolutions.Data.Migrations.csproj:38]: This is part of the file added when I make the update I did not manually alter this

**Comment by sherif-olaboye** [comment]
> Please see inline comment and explain why we need that change. Is that necessary for the work we are doing?

  <Generator>ResXFileCodeGenerator</Generator> This is a generator for resource files (.resx) which generates internal classes for accessing resources.
      <Generator>PublicResXFileCodeGenerator</Generator> This is a generator for resource files (.resx) which generates public classes for accessing resources.
though the change happen by default, I did not manually change it

File: DigitalLearningSolutions.Data.Migrations/202405031044_AlterGetCurrentCoursesForCandidate_V2.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    public class AlterGetCurrentCoursesForCandidate_V2 : Migration
+    {
+        public override void Up()
+        {
+            Execute.Sql(Properties.Resources.TD_3671_Alter_GetCurrentCoursesForCandidate_V2_proc1_up );
+            Execute.Sql(Properties.Resources.TD_3671_Alter_CheckDelegateStatusForCustomisation_func_up );
+        }
+        public override void Down()
+        {
+            Execute.Sql(Properties.Resources.TD_3671_Alter_GetCurrentCoursesForCandidate_V2_proc_down);
```
Comment (rshrirohit) [inline DigitalLearningSolutions.Data.Migrations/202405031044_AlterGetCurrentCoursesForCandidate_V2.cs:15]: There are extra brackets in the down queries. The ones in SSMS and written here do not exactly match. Please see this again.

**Review by rshrirohit — CHANGES_REQUESTED** [review]
Please see inline

**Review by kevwhitt-hee — APPROVED** [review]
